### ISSUE TITLE
[Feature]: Backend support/docs example

### ISSUE THREAD

**Issue by zmzlois** [web-infra-dev/rspress#1151]
### What problem does this feature solve?

wondering when will we incorporate backend support since it's a static site

I was thinking to build an interactive roadmap so people can add features on the doc site using Redis + Deno or whatever edge function but not sure will the deploy fail if I add "use server" flag

example like https://roadmap.upstash.com/ 

### What does the proposed API look like?

a section on the doc about backend: 
1. using serverless function 
2. "use server" whether it is supported? I have see express support on github for rspress, but it's not lived on the docs about how mdx + react component + backend will co-live together

**Comment by chenjiahan** [comment]
Yeah, Rspress is not coupled with any backend solutions, so users can pick anything they want. Also Rspress does not support the "use server" flag.

Examples can be added to https://github.com/rspack-contrib/rspack-examples 😄

**Comment by SoonIter** [comment]
Closing this issue as it has been inactive for over a year. If this is still relevant, please feel free to reopen it or create a new issue with updated context. Thank you for your contribution!
